filters/src/main/java/com/matecat/converter/core/util/Config.java

```diff
@@ -37,6 +37,9 @@
             LOGGER.debug("Loaded configuration:");
 
             serverPort = Integer.parseInt(props.getProperty("server-port"));
+            if (serverPort <= 0) {
+                throw new RuntimeException("The server port specified in the configuration must be positive");
```

Comment:
You should force the check to also have a port number greater than 1024 if the user is not root and **_discourage_** root user `System.getProperty("user.name")`.
The port numbers from 1 to 1023 are restricted for root user only and we can not assign those ports without having root access.
Java app will not start and will crash because it will not be able to bind on the port.

Reply:
OK, I wil add more checks and cross-cheks.

When, and if, we will move to containers this won't necessarily be a problem because of how docker works (the docker daemon runs as root). The MateCat Filters Proxy works this way, it is a container and binds to port 80.